osu.Game/Screens/Edit/Editor.cs

```diff
@@ -394,6 +429,43 @@ private void confirmExit()
             this.Exit();
         }
 
+        private readonly Bindable<string> clipboard = new Bindable<string>();
+
+        protected void Cut()
+        {
+            Copy();
+            foreach (var h in editorBeatmap.SelectedHitObjects.ToArray())
+                editorBeatmap.Remove(h);
+        }
+
+        protected void Copy()
+        {
+            if (editorBeatmap.SelectedHitObjects.Count == 0)
+                return;
+
+            clipboard.Value = new ClipboardContent(editorBeatmap).Serialize();
+        }
+
+        protected void Paste()
+        {
+            if (string.IsNullOrEmpty(clipboard.Value))
+                return;
+
+            var objects = clipboard.Value.Deserialize<ClipboardContent>().HitObjects;
+
+            Debug.Assert(objects.Any());
+
+            double timeOffset = clock.CurrentTime - objects.First().StartTime;
```

Comment:
This `First()` doesn't work in terms of "beatmap time", but in terms of selection order (namely, the first object selected will be pasted in at `CurrentTime` and the rest relative to it, and not necessarily the earliest). Unsure what's the user expectation here; I think I would want timeline order and not selection order but I don't map.

Reply:
wow, select objects should probably definitely be sorted..